pkgs/applications/misc/collision/default.nix

```diff
@@ -0,0 +1,159 @@
+{ lib
+, fetchFromGitHub
+, fetchgit
+, crystal
+, libadwaita
+, gtk4
+, gtk3
+, glib
+, gobject-introspection
+, openssl
+, libxml2
+, pkg-config
+, wrapGAppsHook4
+}:
+
+crystal.buildCrystalPackage rec {
+  pname = "Collision";
+  version = "3.5.0";
+
+  # project source
+  src = fetchFromGitHub {
+    owner = "GeopJr";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    name = pname;
+    hash = "sha256-YNMtiMSzDqBlJJTUntRtL6oXg+IuyAobQ4FYcwOdOas=";
+    };
+
+  # GTK4/libadwaita binding generator for Crystal
+  gi_crystal = fetchFromGitHub {
+      owner = "hugopl";
+      repo = "gi-crystal";
+      rev = "refs/tags/v0.16.0";
+      name = "gi-crystal";
+      hash = "sha256-ij4U8BoSNHpkh5SmvoH5anGB/ZdvmC5zDwJCusR/s/c=";
+    };
+
+  # project dependencies
+  dependencies = [
+    (fetchgit {
+      url = "https://github.com/GeopJr/libadwaita.cr";
+      name = "libadwaita";
+      hash = "sha256-3F1UrApfTQn4kl64PtuQJW/V5XzfToYPqRqT60E2XyY=";
+    })
+    (fetchgit {
+      url = "https://github.com/hugopl/harfbuzz.cr";
+      name = "harfbuzz";
+      hash = "sha256-23CNz9iS9sTw4IYngWq9arH7Kdgqpo/XiDOKom1Z7xA=";
+    })
+    (fetchFromGitHub {
+      owner = "hugopl";
+      repo = "gtk4.cr";
+      rev = "refs/tags/v0.13.0";
+      name = "gtk4";
+      hash = "sha256-deVodUz5dTg/0TxShuesJG2XE+G3c88OTZVvXKBmWXc=";
+    })
+    (fetchFromGitHub {
+      owner = "GeopJr";
+      repo = "non-blocking-spawn";
+      rev = "refs/tags/v1.0.5";
+      name = "non-blocking-spawn";
+      hash = "sha256-6iNwzmRZ84zXxWQ8Z9kcPSmFUBMzuWvcTBNw+g/KL40=";
+    })
+    (fetchFromGitHub {
+      owner = "hugopl";
+      repo = "gio.cr";
+      rev = "refs/tags/v0.1.0";
+      name = "gio";
+      hash = "sha256-qMHPlwaQIoDVcLfTT+gDyoCha3WIZotCtJA0YuIqQ24=";
+    })
+    (fetchFromGitHub {
+      owner = "hugopl";
+      repo = "pango.cr";
+      rev = "refs/tags/v0.2.0";
+      name = "pango";
+      hash = "sha256-TveI7FbFWd4lzV1xUNhs7Iprv36lh/53pX55EWHGgzY=";
+    })
+    (fetchFromGitHub {
+      owner = "GeopJr";
+      repo = "gettext.cr";
+      rev = "refs/tags/v1.0.0";
+      name = "gettext";
+      hash = "sha256-bAY9KevacmclaNP9Ic83EqK/OPz5mSLFKCRncAKpR/g=";
+    })
+    (fetchFromGitHub {
+      owner = "hugopl";
+      repo = "version_from_shard";
+      rev = "refs/tags/v1.2.5";
+      name = "version_from_shard";
+      hash = "sha256-M6L1ZpXNglbf8QKbXSjp7gnmomQE9CV5DqS0TDCQP3Y=";
+    })
+    ];
+
+  sourceRoot = pname;
+
+  # Crystal compiler has a strange issue with OpenSSL. The project will not compile due to
+  # main_module:(.text+0x6f0): undefined reference to `SSL_library_init'
+  # There is an explanation for this https://danilafe.com/blog/crystal_nix_revisited/
+  # Shortly, adding pkg-config to buildInputs along with openssl fixes the issue.
+
+  nativeBuildInputs = [ gtk3 glib gobject-introspection wrapGAppsHook4 ];
+  buildInputs = [ libadwaita gtk4 openssl libxml2 pkg-config ];
```

Comment:
Yes, it has a reason because https://danilafe.com/blog/crystal_nix_revisited/. Without pkg-config in buildInputs package will not build because of crystal compiler error.

Reply:
Again, using `pkg-config` in `buildInputs` is usually wrong, and using it in `nativeBuildInputs` still builds successfully for me, while feeling semantically more correct for potential cross builds.

I sadly can not check your vs my variant in a cross build, as there is LLVM in the dependency graph which needs to be built from scratch for the cross build, which will take a couple of hours on my machine.

Though I will definitely report back once I have more to compare.

Reply:
Hmm, I'll try to build this variant, maybe I got something mixed up and it really builds, sorry

Reply:
It really works, thank you, don't know why i can't build before, but now it's ok.